frida_mode/Makefile

```diff
@@ -0,0 +1,137 @@
+$(FRIDA_BUILD_DIR): | $(BUILD_DIR)
+	mkdir -p $@
+
+$(GUM_DEVKIT_TARBALL): | $(FRIDA_BUILD_DIR)
+	wget -O $@ $(GUM_DEVKIT_URL)
+
+$(GUM_DEVIT_LIBRARY): | $(GUM_DEVKIT_TARBALL)
+	tar Jxvf $(GUM_DEVKIT_TARBALL) -C $(FRIDA_BUILD_DIR)
+
+$(GUM_DEVIT_HEADER): | $(GUM_DEVKIT_TARBALL)
+	tar Jxvf $(GUM_DEVKIT_TARBALL) -C $(FRIDA_BUILD_DIR)
+
+$(FRIDA_TRACE): $(GUM_DEVIT_LIBRARY) $(GUM_DEVIT_HEADER) $(SOURCES) Makefile | $(FRIDA_BUILD_DIR)
+	gcc -shared \
+		-fPIC \
+		-Wno-prio-ctor-dtor \
+		-DWITH_PREFETCH \
+		-DWITH_INLINE_INSTR \
+		-o $@ $(SOURCES) \
+		$(GUM_DEVIT_LIBRARY) \
+		-I $(FRIDA_BUILD_DIR) -I .. -I ../include -I $(INC_DIR) \
+		../instrumentation/afl-compiler-rt.o.c \
+		-lpthread -ldl -lresolv
+
+	cp -v $(FRIDA_TRACE) ../
+
+############################# TEST #############################################
+
+test: $(TEST_BIN)
+
+$(TEST_BUILD_DIR): $(BUILD_DIR)
+	mkdir -p $@
+
+$(HARNESS_FILE): | $(TEST_BUILD_DIR)
+	wget -O $@ $(HARNESS_URL)
+
+$(HARNESS_OBJ): $(HARNESS_FILE)
+	$(CC) -o $@ -c $<
+
+$(PNGTEST_FILE): | $(TEST_BUILD_DIR)
+	wget -O $@ $(PNGTEST_URL)
+
+$(PNGTEST_OBJ): $(PNGTEST_FILE) | $(LIBPNG_DIR)
+	$(CXX) -std=c++11 -I $(LIBPNG_DIR) -o $@ -c $<
+
+$(LIBPNG_FILE): | $(TEST_BUILD_DIR)
+	wget -O $@ $(LIBPNG_URL)
+
+$(LIBPNG_DIR): $(LIBPNG_FILE)
+	tar zxvf $(LIBPNG_FILE) -C $(TEST_BUILD_DIR)
+
+$(LIBPNG_MAKEFILE): | $(LIBPNG_DIR)
+	cd $(LIBPNG_DIR) && ./configure
+
+$(LIBPNG_LIB): $(LIBPNG_MAKEFILE)
+	make -C $(LIBPNG_DIR)
+
+$(TEST_BIN): $(HARNESS_OBJ) $(PNGTEST_OBJ) $(LIBPNG_LIB)
+	$(CXX) \
+		-o $@ \
+		$(HARNESS_OBJ) $(PNGTEST_OBJ) $(LIBPNG_LIB) \
+		-lz
+
+############################# CLEAN ############################################
+clean:
+	rm -rf $(BUILD_DIR)
+
+############################# FORMAT ###########################################
+format:
+	cd .. && echo $(SOURCES) | xargs -L1 ./.custom-format.py -i
+	cd .. && echo $(INCLUDES) | xargs -L1 ./.custom-format.py -i
+
+############################# RUN #############################################
+
+# Add the environment variable AFL_DEBUG_CHILD=1 to show printf's from the target
+
+test_frida: $(FRIDA_TRACE) $(TEST_BIN)
+	make -C ..
+	cd .. && ./afl-fuzz -f frida_mode/frida_test.dat -O -i $(TEST_DATA_DIR) -o frida_mode/frida_out -- frida_mode/build/test/pngtest frida_mode/frida_test.dat
+
+test_qemu: $(TEST_BIN)
+	make -C ..
+	cd .. && ./afl-fuzz -f frida_mode/qemu_test.dat -Q -i $(TEST_DATA_DIR) -o frida_mode/qemu_out -- frida_mode/build/test/pngtest frida_mode/qemu_test.dat
+
+compare: $(FRIDA_TRACE) $(TEST_BIN)
+	cd .. && timeout -s INT 30 ./afl-fuzz -f frida_mode/frida_test.dat -O -i $(TEST_DATA_DIR) -o frida_mode/frida_out -- frida_mode/build/test/pngtest frida_mode/frida_test.dat || true
+	cd .. && timeout -s INT 30 ./afl-fuzz -f frida_mode/qemu_test.dat -Q -i $(TEST_DATA_DIR) -o frida_mode/qemu_out -- frida_mode/build/test/pngtest frida_mode/qemu_test.dat || true
+	cat frida_out/default/fuzzer_stats
```

Comment:
no need for timeout ... just add -V30 to afl-fuzz

Reply:
Fixed